### PR TITLE
Add create vault template command

### LINKED ISSUE #11: Add the function to download the vault template at a later time
### Please agree to the following

- [X] I have searched [existing issues](https://github.com/cryptomator/hub-cli/issues?q=) for duplicates
- [X] I agree to follow this project's [Code of Conduct](https://github.com/cryptomator/hub-cli/blob/develop/.github/CODE_OF_CONDUCT.md)

### Summary

As a Hub-CLI user, I want to be able to download the vault emplate at a later time

### Motivation

Currently you can only download the vault template when you create it. There are scenarios where you want to download it later or download it again.

### Considered Alternatives

_No response_

### Anything else?

hub
├─ vault
│  ├─ template

### PR DESCRIPTION
**Pull request by SailReal**
Fixes #11

## Summary by CodeRabbit

- **New Features**
	- Introduced a command-line tool for creating vault templates, including generating configurations and keys.
	- Added functionality for managing vault configurations with JWT authentication and encryption.
- **Refactor**
	- Streamlined vault creation process by refactoring method calls and removing unused imports.
- **Bug Fixes**
	- Fixed issues related to vault creation and configuration.

### CODE REVIEW

**Comment by SailReal** [comment]
CI fails due to `Error:  nimbus-jose-jwt-9.37.jar: CVE-2023-52428(7.5)`, soon will bump the dependencies during release.

File: src/main/java/org/cryptomator/hubcli/VaultTemplate.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.SecureRandom;
+import java.text.ParseException;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+
+@Command(name = "template", description = "Create vault template of existing vault")
```
Comment (infeo) [inline src/main/java/org/cryptomator/hubcli/VaultTemplate.java]: i suggest to use a verb, e.g. rename it to "download-template" or "create-template".

Comment (infeo) [reply src/main/java/org/cryptomator/hubcli/VaultTemplate.java]: and then also rename the class

File: src/main/java/org/cryptomator/hubcli/VaultTemplate.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			var vaultKeyJWE = backend.getVaultService().getAccessToken(vaultId).body();
+
+			// get device info
+			var device = backend.getDeviceService().get(deviceId);
+
+			// crypto
+			var csprng = SecureRandom.getInstanceStrong();
+			var cliUserPrivateKey = JWEHelper.decryptUserKey(JWEObject.parse(device.userPrivateKey()), deviceKeyPair.getPrivate());
+			try (var vaultKey = JWEHelper.decryptVaultKey(JWEObject.parse(vaultKeyJWE), cliUserPrivateKey)) {
+				try (var configKeyCopy = vaultKey.copy(); var localVaulKeyCopy = vaultKey.copy()) {
```
Comment (infeo) [inline src/main/java/org/cryptomator/hubcli/VaultTemplate.java]: no need for two try-resource-blocks. Do it in one:
```java
try (var vaultKey = JWEHelper.decryptVaultKey(JWEObject.parse(vaultKeyJWE), cliUserPrivateKey);
      var vaultKeyCopy = vaultKey.copy();) {
      //do stuff here
}
```

**Review by infeo — CHANGES_REQUESTED** [review]
I don't like, how we deal with our keys. We only hand over a `key.copy()` into the methods (requires a try-catch-around the method!) and even doing so, sometimes we even make a `keyCopy.copy()`.

But to move fast, i suggest let this technical debt stay. Apart from that, only minor things.

File: src/main/java/org/cryptomator/hubcli/VaultTemplate.java (hunk 1 above)
Comment (SailReal) [reply src/main/java/org/cryptomator/hubcli/VaultTemplate.java]: Thanks for the hint, fixed in cea4aeb293e2bd0b9dd3b7bf99c528d6c4e483b8

File: src/main/java/org/cryptomator/hubcli/VaultTemplate.java (hunk 2 above)
Comment (SailReal) [reply src/main/java/org/cryptomator/hubcli/VaultTemplate.java]: Thanks for the hint, fixed in cea4aeb293e2bd0b9dd3b7bf99c528d6c4e483b8

**Review by infeo — APPROVED** [review]

**Pull request merged**